File: src/main/java/org/jukeboxmc/network/handler/HandlerRegistry.java
package org.jukeboxmc.network.handler;

import com.nukkitx.protocol.bedrock.BedrockPacket;
import com.nukkitx.protocol.bedrock.packet.*;

import java.util.HashMap;
import java.util.Map;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class HandlerRegistry {

    private static final Map<Class<? extends BedrockPacket>, PacketHandler<? extends BedrockPacket>> packetHandlerMap = new HashMap<>();

    public static void init() {
        packetHandlerMap.put( PacketViolationWarningPacket.class, new PacketViolationWarningHandler() );
        packetHandlerMap.put( RequestNetworkSettingsPacket.class, new RequestNetworkSettingsHandler() );
        packetHandlerMap.put( LoginPacket.class, new LoginHandler() );
        packetHandlerMap.put( ResourcePackClientResponsePacket.class, new ResourcePackClientResponseHandler() );
        packetHandlerMap.put( RequestChunkRadiusPacket.class, new RequestChunkRadiusHandler() );
        packetHandlerMap.put( MovePlayerPacket.class, new PlayerMoveHandler() );
        packetHandlerMap.put( TextPacket.class, new TextHandler() );
        packetHandlerMap.put( InventoryTransactionPacket.class, new InventoryTransactionHandler() );
        packetHandlerMap.put( ContainerClosePacket.class, new ContainerCloseHandler() );
        packetHandlerMap.put( InteractPacket.class, new InteractHandler() );
        packetHandlerMap.put( ItemStackRequestPacket.class, new ItemStackRequestHandler() );
        packetHandlerMap.put( MobEquipmentPacket.class, new MobEquipmentHandler() );
        packetHandlerMap.put( AnimatePacket.class, new AnimateHandler() );
        packetHandlerMap.put( PlayerActionPacket.class, new PlayerActionHandler() );
        packetHandlerMap.put( RequestAbilityPacket.class, new RequestAbilityHandler() );
        packetHandlerMap.put( LevelSoundEventPacket.class, new LevelSoundEventHandler() );
        packetHandlerMap.put( BlockPickRequestPacket.class, new BlockPickRequestHandler() );
        packetHandlerMap.put( BlockEntityDataPacket.class, new BlockEntityDataHandler() );
        packetHandlerMap.put( CraftingEventPacket.class, new CraftingEventHandler() );
        packetHandlerMap.put( CommandRequestPacket.class, new CommandRequestHandler() );
        packetHandlerMap.put( AnvilDamagePacket.class, new AnvilDamageHandler() );
        packetHandlerMap.put( RespawnPacket.class, new RespawnHandler() );
        packetHandlerMap.put( EntityEventPacket.class, new EntityEventHandler() );
        packetHandlerMap.put( PlayerSkinPacket.class, new PlayerSkinHandler() );
        packetHandlerMap.put( ModalFormResponsePacket.class, new ModalFormResponseHandler() );
        packetHandlerMap.put( NpcRequestPacket.class, new NpcRequestPacketHandler() );
        packetHandlerMap.put( SetLocalPlayerAsInitializedPacket.class, new SetLocalPlayerAsInitializedHandler() );
        packetHandlerMap.put( EmotePacket.class, new EmoteHandler() );
    }

    public static PacketHandler<? extends BedrockPacket> getPacketHandler( Class<? extends BedrockPacket> clazz ) {
        return packetHandlerMap.get( clazz );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/network/handler/HandlerRegistry.kt
package org.jukeboxmc.server.network.handler

import org.cloudburstmc.protocol.bedrock.packet.*

class HandlerRegistry {

    companion object {
        private val packetHandlerMap: MutableMap<Class<out BedrockPacket>, PacketHandler<out BedrockPacket>> = mutableMapOf()

        init {
            packetHandlerMap[RequestNetworkSettingsPacket::class.java] = RequestNetworkSettingsHandler()
            packetHandlerMap[LoginPacket::class.java] = LoginHandler()
            packetHandlerMap[ClientToServerHandshakePacket::class.java] = ClientToServerHandshakeHandler()
            packetHandlerMap[ResourcePackClientResponsePacket::class.java] = ResourcePackClientResponseHandler()
            packetHandlerMap[ResourcePackChunkRequestPacket::class.java] = ResourcePackChunkRequestHandler()
            packetHandlerMap[RequestChunkRadiusPacket::class.java] = RequestChunkRadiusHandler()
            packetHandlerMap[MovePlayerPacket::class.java] = MovePlayerHandler()
            packetHandlerMap[TextPacket::class.java] = TextHandler()
            packetHandlerMap[EmoteListPacket::class.java] = EmoteListHandler()
            packetHandlerMap[InteractPacket::class.java] = InteractHandler()
            packetHandlerMap[AnimatePacket::class.java] = AnimateHandler()
            packetHandlerMap[PlayerActionPacket::class.java] = PlayerActionHandler()
            packetHandlerMap[ContainerClosePacket::class.java] = ContainerCloseHandler()
            packetHandlerMap[ItemStackRequestPacket::class.java] = ItemStackRequestHandler()
            packetHandlerMap[MobEquipmentPacket::class.java] = MobEquipmentHandler()
            packetHandlerMap[InventoryTransactionPacket::class.java] = InventoryTransactionHandler()
            packetHandlerMap[BlockPickRequestPacket::class.java] = BlockPickRequestHandler()
            packetHandlerMap[PlayerSkinPacket::class.java] = PlayerSkinHandler()
            packetHandlerMap[BlockEntityDataPacket::class.java] = BlockEntityDataHandler()
            packetHandlerMap[LevelSoundEventPacket::class.java] = LevelSoundEventHandler()
            packetHandlerMap[CommandRequestPacket::class.java] = CommandRequestHandler()
            packetHandlerMap[EntityEventPacket::class.java] = EntityEventHandler()
            packetHandlerMap[RespawnPacket::class.java] = RespawnHandler()
            packetHandlerMap[ModalFormResponsePacket::class.java] = ModalFormResponseHandler()
            packetHandlerMap[PlayerAuthInputPacket::class.java] = PlayerAuthInputHandler()
        }

        fun getPacketHandler(clazz: Class<out BedrockPacket?>?): PacketHandler<out BedrockPacket>? {
            return packetHandlerMap[clazz]
        }
    }

}